File: src/main/java/com/manning/javapersistence/ch08/java/model/Item.java
package com.manning.javapersistence.ch08.java.model;

import javax.persistence.*;
import java.util.LinkedHashSet;
import java.util.Set;

@Entity
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(mappedBy = "item", fetch = FetchType.LAZY)
    private final Set<Bid> bids = new LinkedHashSet<>();

    protected Item() {
    }

    public Item(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Set<Bid> getBids() {
        return Set.copyOf(bids);
    }

    public void addBid(Bid bid) {
        bids.add(bid);
    }

    @Override
    public String toString() {
        return "Item{id=" + id + ", name='" + name + "'}";
    }
}

File: src/main/kotlin/com/manning/javapersistence/ch08/kotlin/model/Item.kt
package com.manning.javapersistence.ch08.kotlin.model

import javax.persistence.*

@Entity
class Item(var name: String? = null) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @OneToMany(mappedBy = "item", fetch = FetchType.LAZY)
    private val bids: MutableSet<Bid> = LinkedHashSet()

    fun getBids(): Set<Bid> = bids.toSet()

    fun addBid(bid: Bid) {
        bids.add(bid)
    }

    override fun toString() = "Item{id=$id, name='$name'}"
}
